.pipelines/networkobservability/pipeline.yaml

```diff
@@ -0,0 +1,145 @@
+
+        - task: AzureCLI@1
+          inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                set -e
+                make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=$(clustername)
+          name: "setupkubeconf"
+          displayName: "Set up kubeconfig"
+
+        - script: |
+              CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/main/stable.txt)
+              CLI_ARCH=amd64
+              if [ "$(uname -m)" = "aarch64" ]; then CLI_ARCH=arm64; fi
+              curl -L --fail --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
+              sha256sum --check cilium-linux-${CLI_ARCH}.tar.gz.sha256sum
+              sudo tar xzvfC cilium-linux-${CLI_ARCH}.tar.gz /usr/local/bin
+              rm cilium-linux-${CLI_ARCH}.tar.gz{,.sha256sum}
+          name: "installCiliumCLI"
+          displayName: "Install Cilium CLI"
+
+        - task: AzureCLI@1
+          inputs:
+            azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+            scriptLocation: "inlineScript"
+            scriptType: "bash"
+            addSpnToEnvironment: true
+            inlineScript: |
+              kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/cilium-config/cilium-config-hubble.yaml
+              kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/cilium-agent/files
+              kubectl apply -f test/integration/manifests/cilium/$(cilium_version)/cilium-operator/files
+              envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/$(cilium_version)/cilium-agent/templates/daemonset.tpl | kubectl apply -f -
+              envsubst '${CILIUM_VERSION_TAG},${CILIUM_IMAGE_REGISTRY}' < test/integration/manifests/cilium/$(cilium_version)/cilium-operator/templates/deployment.tpl | kubectl apply -f -
+              # Use different file directories for nightly and current cilium version
+          name: "installCilium"
+          displayName: "Install Cilium on AKS Overlay"
+
+        - script: |
+            echo "Start Azilium E2E Tests on Overlay Cluster"
+            if [ "$CILIUM_VERSION_TAG" = "cilium-nightly-pipeline" ]
+            then
+                CNS=$(CNS_VERSION) IPAM=$(AZURE_IPAM_VERSION) && echo "Running nightly"
+            else
+                CNS=$(make cns-version) IPAM=$(make azure-ipam-version)
+            fi
+            sudo -E env "PATH=$PATH" make test-integration AZURE_IPAM_VERSION=${IPAM} CNS_VERSION=${CNS} INSTALL_CNS=true INSTALL_OVERLAY=true
+          retryCountOnTaskFailure: 3
+          name: "aziliumTest"
+          displayName: "Install Azure-CNS and Run Azilium E2E on AKS Overlay"
+          enabled: true
```

Comment:
Is this pipeline/step also being used in multiple locations?

Reply:
Yes. We need to come up with a strategy here as right now we have duplicate steps/stages scattered across multiple files :-(